lib/direncrypt/inventory.py

```diff
@@ -55,38 +55,103 @@ def read_parameters(self, params_only=False):
                 params[row[0]] = row[1]
         return params
 
-    def read_register(self):
-        """Get information on all registered encrypted files.
+    def read_all_register(self):
+        """Get information on all registered regular files and symlinks.
 
         Returns a dict with unencrypted filename for keys, having
-        a dict of unencrypted file, encrypted file and public id
-        as value.
+        a dict of unencrypted file, encrypted file, public id,
+        is_link and target as value.
         """
         rows = {}
-        for row in self.cursor.execute('''
-                SELECT unencrypted_file, encrypted_file, public_id
-                FROM register'''):
+        request = """SELECT unencrypted_file, encrypted_file, public_id,
```

Comment:
I'd like to avoid repetition here by extracting most of this method with this (or similar) signature:

```
def read_register(self, is_link: bool = None, filtered: str = 'files')     # or filtered='dirs'
```

Reply:
Yes, I corrected this quick-and-dirty code ;-)